app/src/main/java/org/mozilla/fenix/library/bookmarks/toolbar/ToolbarView.kt

```diff
@@ -0,0 +1,117 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.library.bookmarks.toolbar
+
+import android.content.Context
+import androidx.annotation.VisibleForTesting
+import androidx.appcompat.content.res.AppCompatResources
+import androidx.core.content.ContextCompat
+import mozilla.components.browser.toolbar.BrowserToolbar
+import mozilla.components.support.ktx.android.content.getColorFromAttr
+import mozilla.components.support.ktx.android.content.res.resolveAttribute
+import mozilla.components.support.ktx.android.view.hideKeyboard
+import org.mozilla.fenix.R
+import org.mozilla.fenix.library.bookmarks.BookmarkSearchFragmentState
+
+/**
+ * Interface for the Toolbar Interactor. This interface is implemented by objects that want
+ * to respond to user interaction on the [ToolbarView]
+ */
+interface ToolbarInteractor {
+
+    /**
+     * Called when a user removes focus from the [ToolbarView]
+     */
+    fun onEditingCanceled()
+
+    /**
+     * Called whenever the text inside the [ToolbarView] changes
+     * @param text the current text displayed by [ToolbarView]
+     */
+    fun onTextChanged(text: String)
+}
+
+/**
+ * View that contains and configures the BrowserToolbar to only be used in its editing mode.
+ */
+@Suppress("LongParameterList")
+class ToolbarView(
+    private val context: Context,
+    private val interactor: ToolbarInteractor,
+    private val isPrivate: Boolean,
+    val view: BrowserToolbar,
+) {
+
+    @VisibleForTesting
+    internal var isInitialized = false
+
+    init {
+        view.apply {
+            editMode()
+
+            background = AppCompatResources.getDrawable(
+                context, context.theme.resolveAttribute(R.attr.foundation)
+            )
+
+            edit.hint = context.getString(R.string.bookmark_search)
+
+            edit.colors = edit.colors.copy(
+                text = context.getColorFromAttr(R.attr.primaryText),
+                hint = context.getColorFromAttr(R.attr.secondaryText),
```

Comment:
These are `textPrimary` and `textSecondary` now. Probably need to rebase.

Reply:
Thanks, will do.